### DISCUSSION TITLE
Documentation for GitHub Codespaces

### DISCUSSION THREAD

**Discussion by samruddhikhandale** [apache/airflow#29750] [General]
Hi 👋 

I am Samruddhi, a Software Engineer for GitHub, working with the [GitHub Codespaces](https://github.com/features/codespaces) team, and a maintainer for the [devcontainers](https://github.com/devcontainers) org 👩‍💻

I was looking at the [apache/airflow](https://github.com/apache/airflow) repo which has a pretty cool [devcontainer configuration](https://github.com/apache/airflow/tree/main/.devcontainer) setup ✨ 

[GitHub Codespaces](https://docs.github.com/en/codespaces/overview) is a cloud-based development environment that uses your dev container to provide great contributor and maintainer experiences for all users of your project 🪄

🙅‍♀️ No one would ever have to worry about setting up their local machines, adding tools/runtimes for a specific project (which most frequently contradicts with their current environment) and spend hours before even starting their actual work 🙅‍♀️

Would you be willing to accept improvements to your current [devcontainer config](https://github.com/apache/airflow/tree/main/.devcontainer) and a documentation to improve the GitHub Codespaces experience? 💁‍♀️

Thanks,
Samruddhi

// cc @craiglpeters

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
Sure. Just open PR as everyone else here. We are 100% open to new contrbutions, especially those improvig devex. There is no need to ask for it just opening PR  is enough. There is also no need to open issues for that at all.

BTW. I am not sure if you are aware but since you are from GitHub you should be, that there are GitHub discussions that we are heavily using here and in the issue template you are even instructed (and links are presented to you) to open GitHub Discussions if you have questions, rather than opening issues in such cases. 

I realise that the issue here was opened automatically or semi-automatically (I am quite well aware how those things work) but I have a suggestion that instead of opening issues in repositories that use GitHub Discussions, you could improve the automation and open GitHub Discussion instead.

I am converting it discussion where it belongs.

**Reply by samruddhikhandale** [reply]
Awesome, thanks for the information! 😄
